builder/oracle/classic/artifact.go

```diff
@@ -0,0 +1,46 @@
+package classic
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-oracle-terraform/compute"
+)
+
+// Artifact is an artifact implementation that contains a Snapshot.
+type Artifact struct {
+	Snapshot *compute.Snapshot
+	driver   *compute.ComputeClient
+}
+
+// BuilderId uniquely identifies the builder.
+func (a *Artifact) BuilderId() string {
+	return BuilderId
+}
+
+// Files lists the files associated with an artifact. We don't have any files
+// as the custom image is stored server side.
+func (a *Artifact) Files() []string {
+	return nil
+}
+
+func (a *Artifact) Id() string {
+	return a.Snapshot.Name
```

Comment:
Final Artifact should be an [ImageList Entry](https://docs.oracle.com/en/cloud/iaas/compute-iaas-cloud/stcsa/api-ImageListEntries.html) for the created machine image, not the snapshot.  The ImageList and ImageList Entry are the resources needed to enable the created image to be used for launching instances of the image via the UI or APIs. The Snapshot resource can be deleted, it doesn't need to be part of the final artifact.

For the MVP I'd suggest creating a new ImageList and ImageList Entry for each build, a enhancement would be to add a new entry to an existing image list.

Reply:
It turns out that adding a new entry to an existing list is pretty easy; I'll go ahead and add it to this mvp since it's fresh in my mind.